### PR TITLE
ci: improve stability of E2E tests

### LINKED ISSUE #3557: Bug: Repository page throws a hydration error on smaller screens
here's some hydration errors with smaller screens sizes.

CleanShot 2024-06-12 at 16 17 17

### LINKED ISSUE #3946: Chore: Fix Flaky E2E Tests
I've been noticing more frequent flakiness in the E2E test suite, recently. Here is on from a couple minutes ag0, https://github.com/open-sauced/app/actions/runs/10380912089/job/28741490937?pr=3945, but I've also noticed them on releases to main.

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- First pass at improving stability of E2E tests
- Removes checks for URLs for external services
- Fixes a hydration issue on repo pages for small screens
- Increases the timeout for local/CI runs

## Related Tickets & Documents

Closes #3946 
Closes #3557

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: playwright.config.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export default defineConfig({
     /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
     trace: "on-first-retry",
   },
+  timeout: process.env.CI ? 60000 : 30000,
```
Comment (jpmcb) [inline playwright.config.ts:32]: So, the way I read this, if we're in CI, use a longer timeout, otherwise, use a shorter timeout?

**Review by jpmcb — APPROVED** [review]

File: playwright.config.ts (hunk 1 above)
Comment (brandonroberts) [reply playwright.config.ts:32]: Yes

**Review by zeucapua — APPROVED** [review]

File: e2e/repo-page.spec.ts [hunk 2]
```diff
@@ -30,8 +30,7 @@ test.describe("large screen", () => {
     const dialog = await page.getByRole("dialog", { name: "Add repository to Workspace", exact: true });
     await expect(dialog).toBeVisible();
 
-    await dialog.getByRole("button", { name: "Connect with GitHub", exact: true }).click();
-    await expect(page.url()).toContain("https://github.com/login");
+    expect(dialog.getByRole("button", { name: "Connect with GitHub", exact: true })).toBeVisible();
```
Comment (nickytonline) [inline e2e/repo-page.spec.ts:33]: I wonder why clicking it and checking for the Github login was flaky. I'll need to revisit this.

File: e2e/repo-page.spec.ts [hunk 3]
```diff
@@ -53,21 +51,15 @@ test.describe("small screen", () => {
...
 
   test("Adds a repository SBOM to a workspace", async ({ page }) => {
     await page.goto("/s/open-sauced/app");
 
     await page.getByRole("button", { name: "Workspace from SBOM", exact: true }).click();
-    const dialog = await page.getByRole("dialog", { name: "Add repository SBOM to Workspace", exact: true });
+    const dialog = page.getByRole("dialog", { name: "Add repository SBOM to Workspace", exact: true });
     await expect(dialog).toBeVisible();
-
-    await dialog.getByRole("button", { name: "Connect with GitHub", exact: true }).click();
```
Comment (nickytonline) [inline e2e/repo-page.spec.ts:70]: If we remove the remove the click, we can probably still check for the login button in the dialog, but again, need to see what's up with the click not firing consistently to check the login URL.

**Pull request merged**